mlos_bench/mlos_bench/tests/services/remote/azure/azure_services_test.py

```diff
@@ -126,12 +184,46 @@ def test_wait_vm_operation_timeout(mock_requests: MagicMock,
     mock_status_response.json.return_value = {
         "status": "InProgress",
     }
-    mock_requests.get.return_value = mock_status_response
+    mock_session.return_value.get.return_value = mock_status_response
 
     (status, _) = azure_vm_service.wait_host_operation(params)
     assert status == Status.TIMED_OUT
 
 
+@pytest.mark.parametrize(
+    ("total_retries", "operation_status"), [
+        (2, Status.SUCCEEDED),
+        (1, Status.FAILED),
+        (0, Status.FAILED),
+    ])
+@patch("urllib3.connectionpool.HTTPConnectionPool._get_conn")
+def test_wait_vm_operation_retry(mock_getconn: MagicMock,
+                                 total_retries: int,
+                                 operation_status: Status,
+                                 azure_vm_service: AzureVMService) -> None:
+    """
+    Test the retries of the remote VM operation.
+    """
+    params = {
+        "pollInterval": 0.1,
+        "requestTotalRetries": total_retries,
+        "asyncResultsUrl": "https://DUMMY_ASYNC_URL",
+        "vmName": "test-vm",
+    }
+
+    # Simulate intermittent connection issues
+    mock_getconn.return_value.getresponse.side_effect = [
+        make_httplib_json_response(200, {"status": "InProgress"}),
+        requests_ex.ConnectionError("Connection aborted", OSError(107, "Transport endpoint is not connected")),
+        requests_ex.ConnectionError("Connection aborted", OSError(107, "Transport endpoint is not connected")),
```

Comment:
repeat?

Reply:
Yeah I intended for this repeat to test cases of multiple retry attempts.
For example having >= 2 maximum retries should result in success, but having 1 maximum retry should still have a graceful failure.

Reply:
Maybe just add a comment explaining that.  Thanks!

Reply:
Added the comment!